File: frontend/src/models/LinkPerformance.ts
import type { CrawlingLinkSource } from '@/models/CrawlingLinkSource'

export class LinkPerformance {
  public url: string;
  public timeResponseMs: number;
  public crawlingLinkSource: CrawlingLinkSource;
  constructor(url: string, timeResponseMs: number, crawlingLinkSource: number) {
    this.url = url;
    this.timeResponseMs = timeResponseMs;
    this.crawlingLinkSource = crawlingLinkSource;
  }

}

File: WebsitePerformanceEvaluator.Crawler/Models/LinkPerformance.cs
using WebsitePerformanceEvaluator.Crawler.Models.Enums;

namespace WebsitePerformanceEvaluator.Crawler.Models;

public class LinkPerformance
{
    public string Url { get; set; }
    public int? TimeResponseMs { get; set; }
    public CrawlingLinkSource CrawlingLinkSource { get; set; }
    public override int GetHashCode()
    {
        return Url.GetHashCode();
    }
    
    public override bool Equals(object? obj)
    {
        return obj is LinkPerformance link && link.Url == Url;
    }
}